modules/Makefile.in

```diff
@@ -62,7 +62,7 @@ C=
 endif
 
 ifneq "$(LIBZNC)" ""
-LIBS += -L.. -lznc -Wl,-rpath,$(LIBZNCDIR)
+LIBS += -L.. -lznc -lz -Wl,-rpath,$(LIBZNCDIR)
```

Comment:
1. If you're changing Makefile.in, probably CMakeLists.txt also needs to be updated
2. This line would work only for Cygwin, as other OSes don't use $LIBZNC
3. ZNC already checks whether zlib is available, and turns on its support automatically (for HTTP server)

Just revert this.